core/src/main/java/org/axonframework/serialization/MessageSerializer.java

```diff
@@ -60,9 +60,6 @@ public MessageSerializer(Serializer serializer) {
      */
     public static <T> SerializedObject<T> serializePayload(Message<?> message, Serializer serializer,
                                                            Class<T> expectedRepresentation) {
-        if (message instanceof SerializationAware) {
```

Comment:
I think we can remove the entire `MessageSerializer` class, now. This logic should be part of the message implementations itself.